src/filesystem/fs.rs

```diff
@@ -1,17 +1,22 @@
-use crate::errors::{Error, Result};
-use crate::filesystem::binding::Side::Host;
-use crate::filesystem::binding::{Binding, Side};
-use nix::sys::stat::Mode;
+use std::ffi::CString;
 use std::fs::Metadata;
 use std::path::{Path, PathBuf};
 
+use nix::sys::{self, stat::Mode};
+use nix::unistd::{self, AccessFlags};
+
+use crate::errors::{Error, Result, WithContext};
+use crate::filesystem::binding::Side::Host;
+use crate::filesystem::binding::{Binding, Side};
+
 /// Information related to a file-system name-space.
 #[derive(Debug)]
 pub struct FileSystem {
     /// List of bindings used to replicate `mount` and `bind`.
     /// It will also contain the root binding (to replicate `chroot`).
     bindings: Vec<Binding>,
-    /// Working directory in gusetfs, à la `/proc/self/pwd`.
+    /// Working directory in gusetfs, à la `/proc/self/cwd`. Always is absolute
```

Comment:
```suggestion
    /// Working directory in guestfs, e.g., `/proc/self/cwd`, is always absolute
```

Reply:
Do we have a unit test for this?

Reply:
> Do we have a unit test for this?

Not yet. This part is too coupled and I haven't come up with a suitable way to write unit tests.